### DISCUSSION TITLE
After `git rebase`, I have no contribution, but I am still in Contributors

### DISCUSSION THREAD

**Discussion by BaldStrong** [community/community#73032] [Repositories]
### Select Topic Area

Bug

### Body
**I previously contributed to this repository using my own account, but after a git rebase, I can no longer find any trace of my contributions. Why am I still listed in the Contributors? Could this be a bug on GitHub?**

**Comment by EeyoreLee** [comment]
I think the revert commit still trace your commit, so there's a contribution

**Reply by BaldStrong** [reply]
Hello EeyoreLee, 
Thank you very much for your explanation. I've discovered that there are indeed dangling commits in the repository, but they are not present in the cloned repository. It's necessary to manually execute `git fetch <dangling commit id>` to have them show up in the results of `git fsck --lost-found --unreachable --dangling --full --name-objects`. However, after performing a local `git gc` and then trying to `git push`, it states `Everything up-to-date`. Is there a way to remove dangling commits from a GitHub remote repository? I've heard that GitHub periodically cleans up these dangling commits, but several months have passed and they are still there.

**Comment by Yakkhini** [comment]
https://github.com/orgs/community/discussions/49813

This may helpful to you. Change the main branch name, refresh the contributor list, then change back. After that, dangling commit / legacy commits that have been rebased could be ignored.

**Comment by deven1003** [comment]
This isn't a bug it's expected Git and GitHub behavior. A few things are likely happening:

**Author vs. committer.** Every commit stores two identities: the *author* (who originally wrote it) and the *committer* (who applied it). A `git rebase` preserves the original **author** but rewrites the **committer** to whoever ran the rebase. GitHub attributes contributions to the *author*, so your name remains in the Contributors list even after the rebase.

**New commit SHAs.** Rebasing rewrites history and assigns your commits entirely new SHAs. That's why you "can't find" your original commits — they may still exist under different IDs with your authorship intact.

**Possible squashing.** If the rebase squashed multiple commits into one (or the PR was squash-merged), only a single author is credited on the combined commit, so individual contributions can disappear from the visible history. This is by design.

**Caching.** GitHub recomputes the contributor graph on a delay after a force-push, so your name can persist there for a while even if the underlying commits changed.

You can verify how your identity is attached by running:

```
git log --all --author="your-email@example.com" --format="%h | author: %an | committer: %cn"
```
